### DISCUSSION TITLE
使用上遇到的问题

### DISCUSSION THREAD

**Discussion by wagcheg** [xmake-io/xmake#5862] [Ideas]
flamegraph 处理xmake性能的性能问题
add_deps(, {public = false}) 动态库包装静态库不引入额外包含目录和宏等属性
msvc不需要映射std:c++11 qt5规则默认带上了std::c++11
vsxmake里其他类型的文件也要能在工程里
ninja生成器phony target
phony target不能应用rules
vsxmake生成器 object target报错
cmake生成器 mainfest pch
qt rules 三方静态库扫描未缓存 很多target时config慢
moc rules 在未发生更改的生成中耗时高
sharedpch

**Comment by star-hengxing** [comment]
> add_deps(, {public = false})

没这种用法，如果是运行时依赖就 `add_deps(, {inherit = false})`

> msvc不需要映射std:c++11 qt5规则默认带上了std::c++11

新的 msvc 只支持 >= c++14 ，-std:c++11 会自动过滤。

> vsxmake生成器 object target报错

2.9.6 修了。

> sharedpch

https://github.com/xmake-io/xmake/issues/4167
